crates/nu-command/src/filesystem/start.rs

```diff
@@ -130,3 +133,47 @@ impl Command for Start {
         ]
     }
 }
+
+fn open_path(
+    path: impl AsRef<std::ffi::OsStr>,
+    engine_state: &EngineState,
+    stack: &Stack,
+    span: Span,
+) -> Result<(), ShellError> {
+    try_commands(open::commands(path), engine_state, stack, span)
+}
+
+fn try_commands(
+    commands: Vec<std::process::Command>,
+    engine_state: &EngineState,
+    stack: &Stack,
+    span: Span,
+) -> Result<(), ShellError> {
+    let env_vars_str = env_to_strings(engine_state, stack)?;
+    commands
+        .into_iter()
+        .map(|mut cmd| {
+            let status = cmd
+                .envs(&env_vars_str)
+                .stdin(Stdio::null())
+                .stdout(Stdio::null())
+                .stderr(Stdio::null())
+                .status();
+            match status {
+                Ok(status) if status.success() => Ok(()),
+                Ok(status) => Err(format!("\nCommand {cmd:?} failed with {:?}", status)),
+                Err(err) => Err(format!("\nCommand {cmd:?} failed with {:?}", err)),
+            }
+        })
+        .take_while_inclusive(|result| result.is_err())
+        .fold(Err("".to_string()), |combined_result, next_result| {
+            combined_result.or_else(|combined_message| {
+                next_result.map_err(|next_message| combined_message + &next_message)
+            })
+        })
+        .map_err(|message| ShellError::ExternalCommand {
+            label: "No command found to start with this path".to_string(),
+            help: "Try different path or install appropriate command\n".to_string() + &message,
+            span,
+        })
```

Comment:
My first impression was that this makes it harder for the user to see if there is a problem with the path or URL or if there is no application linked succesfully to the type.  
The information would be written out in the long string if there were multiple attempts. As much as possible we try to use the explicit common `ShellError` variants to express what went wrong.

Reply:
For each failed command there will be an output in a separate line.
With regards to explicitness, is ExternalCommand not explicit enough? I'm not sure I follow what are you saying.

Reply:
Yeah the previous output also wasn't any more helpful and it is fully up to the behavior of the external command, so we can probably keep it for now, as long as it is reasonably readable.